File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    
    stack = []
    
    while prices:
        # 가장 마지막 원소 pop
        p = prices.pop()
        
        if len(answer) == 0: # 아무 원소도 들어있지 않을 경우 ( 가장 마지막 )
            answer.append(0)
            stack.append(p)
        elif len(answer) == 1: # 원소가 하나밖에 없을 경우
            answer.append(1)
            stack.append(p)
        else:
            cnt = 1
            check = False
            # 스택의 마지막부터 비교
            for i in range(len(stack)-1, -1, -1):
                if p <= stack[i]:
                    cnt += 1
                else:
                    answer.append(cnt)
                    check = True
                    break
            # 마지막까지 stack 을 반복했을경우
            if not check:
                answer.append(cnt-1)
            
            stack.append(p)
            
    answer.reverse()
    
    return answer

"""
## 주식 가격

## 문제
1. 초 단위로 기록된 주식 가격이 담긴 배열

2. 가격이 떨어지지 않은 기간은 몇 초인지 return 하는 함수 만들기

## 풀이
1. prices 뒤에서 접근

2. 
"""

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public List<Integer> solution(int[] prices) {
        List<Integer> answer = new ArrayList<>();
        
        ArrayDeque<Integer> queue = new ArrayDeque<>();
        
        for(int price : prices) {
            queue.add(price);
        }
        
        while (!queue.isEmpty()) {
            int p = queue.pollFirst();
            int cnt = 0;
            for (int q : queue) {
                if (p <= q) {
                    cnt++;
                } else {
                    cnt++;
                    break;
                }
            }
            answer.add(cnt);
            
        }
        
        return answer;
    }
}